itest/lnd_coop_close_external_delivery_test.go

```diff
@@ -53,20 +109,102 @@ var coopCloseWithExternalTestCases = []*lntest.TestCase{
 // balance irrespective of whether the delivery address is in LND's wallet or
 // not. Some users set this value to be an address in a different wallet and
 // this should not affect our ability to accurately report the settled balance.
+//
+// If importTapscript is set, it imports a Taproot script and internal key to
+// alice LND using ImportTapscript to make sure it doesn't interfere with
+// delivery address. See https://github.com/lightninglabs/loop/issues/923
+// If importPubkey is set, the address is imported using ImportPublicKey.
 func testCoopCloseWithExternalDelivery(ht *lntest.HarnessTest,
-	upfrontShutdown bool, deliveryAddressType lnrpc.AddressType) {
+	upfrontShutdown, importTapscript, importPubkey bool,
+	deliveryAddressType lnrpc.AddressType) {
 
 	alice := ht.NewNodeWithCoins("Alice", nil)
 	bob := ht.NewNodeWithCoins("bob", nil)
 	ht.ConnectNodes(alice, bob)
 
+	// Make fake taproot internal public key (equal to the final).
+	// This is only used if importTapscript or importPubkey is set.
+	taprootPubkey := [32]byte{1, 2, 3}
+	if upfrontShutdown {
+		// Make new address for second sub-test not to import
+		// the same address twice causing an error.
+		taprootPubkey[3] = 1
+	}
+	pkScriptBytes := append(
+		[]byte{txscript.OP_1, txscript.OP_DATA_32},
+		taprootPubkey[:]...,
+	)
+	pkScript, err := txscript.ParsePkScript(pkScriptBytes)
+	require.NoError(ht, err)
+	taprootAddress, err := pkScript.Address(harnessNetParams)
+	require.NoError(ht, err)
+
+	var addr string
+	switch {
+	// Use ImportTapscript.
+	case importTapscript:
+		addr = taprootAddress.String()
+
+		// Import the taproot address to LND.
+		req := &walletrpc.ImportTapscriptRequest{
+			InternalPublicKey: taprootPubkey[:],
+			Script: &walletrpc.ImportTapscriptRequest_FullKeyOnly{
+				FullKeyOnly: true,
+			},
+		}
+		res := alice.RPC.ImportTapscript(req)
+		require.Equal(ht, addr, res.P2TrAddress)
+
+	// Use ImportPublicKey.
+	case importPubkey:
+		var (
+			address     btcutil.Address
+			pubKey      []byte
+			addressType walletrpc.AddressType
+		)
+		switch deliveryAddressType {
+		case lnrpc.AddressType_UNUSED_WITNESS_PUBKEY_HASH:
+			// Make fake public key hash.
+			pk := [33]byte{2, 3, 4}
+			if upfrontShutdown {
+				// Make new address for second sub-test.
+				pk[1]++
```

Comment:
Is this actually doing anything? There's no re-assignment here.

Reply:
Ah TIL, kind of surprising behavior: https://go.dev/play/p/W07kpFn0B7M

Reply:
Yeah, this operation changes the value in the array in-place.